gcs/src/main/scala/monix/connect/gcp/storage/components/GcsUploader.scala

```diff
@@ -46,44 +45,61 @@ private[storage] final class GcsUploader(
   override def createSubscriber(
     cb: Callback[Throwable, Unit],
     s: Scheduler): (Subscriber[Array[Byte]], AssignableCancelable) = {
-    val sub = new Subscriber[Array[Byte]] {
+    val sub = new Subscriber.Sync[Array[Byte]] {
       self =>
       override implicit def scheduler: Scheduler = s
 
       val writer: WriteChannel = storage.writer(blobInfo, options: _*)
       writer.setChunkSize(chunkSize)
+      private[this] var isDone = false
 
-      override def onNext(chunk: Array[Byte]): Future[Ack] = {
-        Task {
+      override def onNext(chunk: Array[Byte]): Ack = {
+        try {
```

Comment:
What is this `try` protecting against?

If `writer.write` can throw then we could only wrap that part of the code

Reply:
mmh yes that would be more accurate :)